### PR TITLE
runAction (non streamed) + typing typing typing

### PR DESCRIPTION
**Pull request by spolu**
This PR cleans up a and strictens a lot of types. It also introduce a non streamed `runAction` that should be preferred for Doc Tracker and Events over runApp (it's an observable wrapper on top of it) cc @fontanierh @PopDaph

### CODE REVIEW

**Comment by spolu** [comment]
@fontanierh unknown FTW
cc @philipperolet @Uzay-G for impact on chat and gens

File: front/lib/actions/server.ts [hunk 1]
```diff
@@ -96,3 +110,63 @@ export async function runActionStreamed(
 
   return new Ok({ eventStream: streamEvents(), dustRunId });
 }
+
+export async function runAction(
```
Comment (spolu) [inline front/lib/actions/server.ts:114]: The new `runAction`

File: front/lib/dust_api.ts [hunk 2]
```diff
@@ -1,35 +1,16 @@
...
   message: string;
 };
-export type DustAPIResponse<T> = Result<T, DustAPIErrorResponse>;
-
-export type DustAPINonStreamedResponse = Result<
-  DustAPINonStreamedResponseSuccess,
-  DustAPIErrorResponse
->;
-
-export type DustAPINonStreamedResponseSuccess = {
```
Comment (spolu) [inline front/lib/dust_api.ts:20]: ==> RunType which pre-existed

File: front/lib/dust_api.ts [hunk 3]
```diff
@@ -146,23 +127,7 @@ export type DustAPICredentials = {
  *
  * @param res an HTTP response ready to be consumed as a stream
  */
-export async function processStreamedRunResponse(res: Response): Promise<
-  DustAPIResponse<{
```
Comment (spolu) [inline front/lib/dust_api.ts:150]: I figured no need to explicit these types they are inferred by the type system

File: front/types/run.ts [hunk 4]
```diff
@@ -36,9 +36,9 @@ export type BlockStatus = {
 };
 
 export type TraceType = {
-  value?: any;
-  error?: string;
-  meta?: any;
+  value: unknown | null;
```
Comment (spolu) [inline front/types/run.ts:39]: we are receiving `null` not undefined from the network I verified + move to unknown, all in all better typing

**Review by PopDaph — APPROVED** [review]
LGTM! 
Typing makes it less easy to read but it feels more comfortable as is 👍🏻

**Pull request merged**